### PR TITLE
Symbolic Execution refactoring

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/SymbolicValues/NullableSymbolicValue.cs [hunk 1]
```diff
@@ -68,31 +66,18 @@ public NullableSymbolicValue(SymbolicValue wrappedValue)
...
-        public override string ToString()
-        {
-            if (base.identifier != null)
-            {
-                return $"NULLABLE_SV_{base.identifier}";
-            }
-
-            return WrappedValue?.ToString() ?? base.ToString();
-        }
+        public override string ToString() => $"NULLABLE_SV_{identifier}" ;
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/SymbolicValues/NullableSymbolicValue.cs:81]: The previous version displayed the `WrappedValue` when the `identifier` was null. This is no longer needed?

**Review by costin-zaharia-sonarsource — COMMENTED** [review]
Just one comment regarding the NullableSymbolicValue.ToString change.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/SymbolicValues/NullableSymbolicValue.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/SymbolicValues/NullableSymbolicValue.cs:81]: There's no way to make `identifier` null. The `identifier` is set in base class constructor (it's the counter). This class cannot break it unless new constructor with `: base(null)` is added. So it was useless anyway and could be removed.

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**